### ISSUE TITLE
Preload, background loading and service worker

### ISSUE THREAD

**Issue by clagom** [01org/zephyrjs-ide#93]
**Issue description**
Instead of having a loading, we could change to load the necessary elements for the view first, and then load the rest in background, using service worker for caching.
https://medium.com/dev-channel/preload-prefetch-and-priorities-in-chrome-776165961bbf 
CC @kenchris 

**Current behavior**
Loading spinner for the whole page

**'JS IDE for Zephyr OS' version:** `X.Y.Z / abcd1234`

**Comment by kenchris** [comment]
Yes this is very important, at least if you don't want to get bashed by very vocal web developers in high places on social media :-)

There is no reason that the front page has a spinner at all. It is one big image and some text basically and other sites can load and show that immediately. The background picture can probably be compressed better (Google just showed how to compress JPEGs around 30% more) and it could load lazily and fade in when ready. And for sequential loads, having it cached by the Service Worker would make it load instantly.
